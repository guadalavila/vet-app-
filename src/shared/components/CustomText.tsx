import React, { useContext } from 'react';
import { StyleProp, StyleSheet, Text as TextRN, TextStyle } from 'react-native';
import { ThemeContext } from '~contexts/ThemeContext';

interface ICustomTextProps {
    children: React.ReactNode;
    style?: StyleProp<TextStyle> | undefined;
    props?: any;
    testID?: string | undefined;
}
const CustomText = ({ children, style = {}, props = {}, testID }: ICustomTextProps) => {
    const { themeApp } = useContext(ThemeContext);

    return (
        <TextRN testID={testID} {...props} style={[{ color: themeApp.colors.text }, style]}>
            {children}
        </TextRN>
    );
};

export default CustomText;

const styles = StyleSheet.create({});
