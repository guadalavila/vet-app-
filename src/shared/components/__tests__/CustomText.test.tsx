import { render, screen } from '@testing-library/react-native';
import React from 'react';
import CustomText from '../CustomText';

describe('CustomText', () => {
    test('should match snapshot', () => {
        const customText = render(<CustomText testID='custom-text'>Hello World!</CustomText>);
        expect(customText).toMatchSnapshot();
    });
    test('should render component', () => {
        render(<CustomText testID='custom-text'>Hello World!</CustomText>);
        expect(screen.getByTestId('custom-text')).toBeTruthy();
        expect(screen.getByTestId('custom-text').children).toEqual(['Hello World!']);
    });
});
