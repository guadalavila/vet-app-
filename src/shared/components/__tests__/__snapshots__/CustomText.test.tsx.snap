// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomText should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#11100F",
        "fontSize": 16,
      },
      Object {},
    ]
  }
  testID="custom-text"
>
  Hello World!
</Text>
`;
